.github/workflows/cibuildwheel.yml

```diff
@@ -90,12 +90,35 @@ jobs:
     name: Build wheels on ${{ matrix.os }} for ${{ matrix.cibw_archs }}
     runs-on: ${{ matrix.os }}
     env:
-      CIBW_BEFORE_BUILD: >-
-        pip install numpy>=1.25 &&
-        rm -rf {package}/build
-      CIBW_BEFORE_BUILD_WINDOWS: >-
-        pip install delvewheel numpy>=1.25 &&
-        rm -rf {package}/build
+      # The following commands branch based on whether we are on the main
+      # branch or have a PR into main.  For these, we use the NumPy 2.0
+      # nightlies to ensure future C-API/ABI compatibility.
+      # This branching becomes unnecessary when NumPy 2.0 is released.
+      # When using no-build-isolation we need to install all requirements.
+      CIBW_BEFORE_BUILD:
+        ${{ (((github.event_name == 'push' && github.ref == 'refs/heads/main') ||
+              (github.event_name == 'pull_request' && github.base_ref == 'refs/heads/main')) &&
+            'pip install meson-python pybind11 setuptools_scm "setuptools>=64" ninja &&
+             pip install "numpy>=2.0.0.dev0" --extra-index-url "https://pypi.anaconda.org/scientific-python-nightly-wheels/simple" &&
+             rm -rf {package}/build'
+          ) ||
+            'pip install "numpy>=1.25" &&
+             rm -rf {package}/build'
+        }}
+      CIBW_BEFORE_BUILD_WINDOWS:
+        ${{ (((github.event_name == 'push' && github.ref == 'refs/heads/main') ||
+              (github.event_name == 'pull_request' && github.base_ref == 'refs/heads/main')) &&
+            'pip install delvewheel meson-python pybind11 setuptools_scm "setuptools>=64" ninja &&
+             pip install "numpy>=2.0.0.dev0" --extra-index-url "https://pypi.anaconda.org/scientific-python-nightly-wheels/simple" &&
+             rm -rf {package}/build'
+          ) ||
+            'pip install delvewheel "numpy>=1.25" &&
+             rm -rf {package}/build'
+        }}
+      CIBW_BUILD_FRONTEND: >-
+        ${{ (((github.event_name == '' && github.ref == 'refs/heads/main') ||
```

Comment:
```suggestion
        ${{ (((github.event_name == 'push' && github.ref == 'refs/heads/main') ||
```

?

Reply:
ouch, I was testing both branches, and forgot to undo it :facepalm: